lib/flatpak-installation.c

```diff
@@ -847,21 +858,30 @@ flatpak_installation_list_installed_refs_by_kind (FlatpakInstallation *self,
  * it can have local updates available that has not been deployed. Look
  * at commit vs latest_commit on installed apps for this.
  *
- * Returns: (transfer container) (element-type FlatpakInstalledRef): an GPtrArray of
- *   #FlatpakInstalledRef instances
+ * Returns: (transfer container) (element-type FlatpakInstalledRef): a GPtrArray of
+ *   #FlatpakInstalledRef instances, or %NULL on error
  */
 GPtrArray *
 flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
                                                      GCancellable        *cancellable,
                                                      GError             **error)
 {
-  g_autoptr(GPtrArray) updates = NULL;
-  g_autoptr(GPtrArray) installed = NULL;
-  g_autoptr(GPtrArray) remotes = NULL;
-  g_autoptr(GHashTable) ht = NULL;
+  g_autoptr(GPtrArray) updates = NULL; /* (element-type FlatpakInstalledRef) */
+  g_autoptr(GPtrArray) installed = NULL; /* (element-type FlatpakInstalledRef) */
+  g_autoptr(GPtrArray) remotes = NULL; /* (element-type FlatpakRemote) */
+  g_autoptr(GHashTable) remote_commits = NULL; /* (element-type utf8 utf8) */
   int i, j;
+#ifdef FLATPAK_ENABLE_P2P
+  g_autoptr(FlatpakDir) dir = NULL;
+  g_autoptr(GMainContext) context = NULL;
+  g_auto(OstreeRepoFinderResultv) results = NULL;
+  g_autoptr(GAsyncResult) result = NULL;
+  GPtrArray *collection_refs = NULL; /* (element-type OstreeCollectionRef) */
```

Comment:
This is not leaked, but you're properly using g_steal_pointer on it, so its probably nicer to make it g_autoptr.